float64 target_width
---
bool success
